Update to ACE and TAO for the current version 8.0.0

lockhart opened the pull request:
#### Description

<!-- Note: it is best to make pull requests from a branch rather than from master -->

###### Type(s)
<!-- update (title contains ": U(u)pdate to"), submission (new Portfile) and CVE Identifiers are auto-detected, replace [ ] with [x] to select -->

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.5 23F79 arm64
Xcode 15.4 15F31d / Command Line Tools 15.3.0.0.1.1708646388

###### Verification <!-- (delete not applicable items) -->
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

<!-- Use "skip notification" (surrounded with []) to avoid notifying maintainers -->

ryandesign commented on devel/ace/Portfile:
```diff
@@ -72,9 +77,6 @@ foreach {key value} [array get os.names]  {
 set subdirs         { ace ACEXML ASNMP Kokyu apps netsvcs }
 
 post-patch {
-    reinplace "s|@MACOSX@|${os.name}|g" \
-        ${worksrcpath}/ace/config.h \
-        ${worksrcpath}/include/makeinclude/platform_macros.GNU
```
Since the `os.name` variable is now unused in both ports, can the code that sets that variable be removed from both ports?

lockhart commented:
Yes that can be removed.

My version of the Portfile in https://github.com/lockhart/macports-ports/blob/ace_tao-8.0.0/devel/ace/Portfile which should have been included in the pull request does have it removed. The next-most-recent version of the file in git is 3 years ago.

I’m not sure how to verify in hindsight what file you see in the pull request; advice on how to help is appreciated.

Thanks!

- Tom

> On May 23, 2024, at 2:04 PM, Ryan Carsten Schmidt ***@***.***> wrote:
> 
> 
> @ryandesign commented on this pull request.
> 
> In devel/ace/Portfile <https://github.com/macports/macports-ports/pull/24095#discussion_r1612303769>:
> 
> > -    reinplace ***@***.***@|${os.name}|g" \
> -        ${worksrcpath}/ace/config.h \
> -        ${worksrcpath}/include/makeinclude/platform_macros.GNU
> Since the os.name variable is now unused in both ports, can the code that sets that variable be removed from both ports?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/macports/macports-ports/pull/24095#pullrequestreview-2075022968>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AANLDAKG623L5VFQBB37X5LZDZKV7AVCNFSM6AAAAABIGFXGT2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDANZVGAZDEOJWHA>.
> You are receiving this because you authored the thread.
>

lockhart commented:
I’ve made two new commits to remove the remaining unused code in the devel/ace and devel/tao Portfiles.

It seems that this has been integrated into the pull request by github but let me know if you need me to do something else.

Thanks!

- Tom

> On May 23, 2024, at 2:04 PM, Ryan Carsten Schmidt ***@***.***> wrote:
> 
> 
> @ryandesign commented on this pull request.
> 
> In devel/ace/Portfile <https://github.com/macports/macports-ports/pull/24095#discussion_r1612303769>:
> 
> > -    reinplace ***@***.***@|${os.name}|g" \
> -        ${worksrcpath}/ace/config.h \
> -        ${worksrcpath}/include/makeinclude/platform_macros.GNU
> Since the os.name variable is now unused in both ports, can the code that sets that variable be removed from both ports?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/macports/macports-ports/pull/24095#pullrequestreview-2075022968>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AANLDAKG623L5VFQBB37X5LZDZKV7AVCNFSM6AAAAABIGFXGT2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDANZVGAZDEOJWHA>.
> You are receiving this because you authored the thread.
>

jwillemsen commented on devel/ace/files/patch-include-makeinclude-platform_macros.GNU.diff:
```diff
@@ -1,6 +1,7 @@
 +++ include/makeinclude/platform_macros.GNU
-@@ -0,0 +1,12 @@
+@@ -0,0 +1,13 @@
++c++std=c++17
```
Isn't that done already in platform_macosx_common.GNU?

lockhart commented:
Not sure why but I had thought that the build failed until I added this line. More likely, I added it when upgrading my own software to this newest version of TAO and then did not revert back when I had learned more about what was required.

After testing without, this line is unnecessary as Jonny points out.

It does not hurt the macports build on clang (which is likely a sufficient configuration in case it leaks through into the released build) but I’ve removed it from both ACE and TAO in the github pull request.

Thanks!

- Tom

> On May 25, 2024, at 12:29 AM, Johnny Willemsen ***@***.***> wrote:
> 
> 
> @jwillemsen commented on this pull request.
> 
> In devel/ace/files/patch-include-makeinclude-platform_macros.GNU.diff <https://github.com/macports/macports-ports/pull/24095#discussion_r1614424421>:
> 
> > @@ -1,6 +1,7 @@
>  --- include/makeinclude/platform_macros.GNU.orig
>  +++ include/makeinclude/platform_macros.GNU
> -@@ -0,0 +1,12 @@
> +@@ -0,0 +1,13 @@
> ++c++std=c++17
> Isn't that done already in platform_macosx_common.GNU?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/macports/macports-ports/pull/24095#pullrequestreview-2078813646>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AANLDAM5M2MCK464W6PRHKDZEA4VNAVCNFSM6AAAAABIGFXGT2VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDANZYHAYTGNRUGY>.
> You are receiving this because you authored the thread.
>

ryandesign commented:
Thanks. All that remains is to squash all the ace-related commits and all the tao-related commits into a single commit each and to write a single consolidated commit message for each. I can do that later tonight or tomorrow if nobody beats me to it.

lockhart commented:
I haven’t squashed before, so don’t want to screw it up.

fwiw both ACE and TAO can have the same commit message since the nature of the changes are the same. If you end up doing the squash across them both, the commit message could be:

Update to version 8.0.0.
Remove the @macosx@ file name substitution since the distro now does this internally.
Remove obsolete code which determined how to name include files.

Thanks for the help!

- Tom

> On May 25, 2024, at 11:25 AM, Ryan Carsten Schmidt ***@***.***> wrote:
> 
> 
> Thanks. All that remains is to squash all the ace-related commits and all the tao-related commits into a single commit each and to write a single consolidated commit message for each. I can do that later tonight or tomorrow if nobody beats me to it.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.Message ID: ***@***.***>

pmetzger commented:
@lockhart Squashing commits is not difficult. I recommend just reading online how to do it. Basically you just want to cherry pick, rebase, force push but you should read up on it. You can always copy your repo to practice fwiw.

lockhart commented:
The first try at squashing went very poorly. Will try one of the other dozens of suggested recipes as time allows.

- Tom

> On May 27, 2024, at 12:07 PM, Perry E. Metzger ***@***.***> wrote:
> 
> 
> @lockhart <https://github.com/lockhart> Squashing commits is not difficult. I recommend just reading online how to do it. Basically you just want to cherry pick, rebase, force push but you should read up on it. You can always copy your repo to practice fwiw.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/macports/macports-ports/pull/24095#issuecomment-2133949271>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AANLDALYV3HUI4ABNFP2P5LZEN77LAVCNFSM6AAAAABIGFXGT2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCMZTHE2DSMRXGE>.
> You are receiving this because you were mentioned.
>

jwillemsen commented:
Or copy the changes files to a tmp dir, make a new bracht and make one commit with all changes files and open a new PR

pmetzger commented:
I find that generally, the reason people have trouble squashing commits is that they don't have a good model in their mind of how Git works. I recommend reading a description of git's internals (like in the online book Pro Git); once you get how git works, the goal of rebasing (or the like) becomes much more obvious. The command syntax is still often terrible, mind you, but you know what you're trying to do.

lockhart commented:
Well duh. Yeah I’ll do that. Probably will label the pull request as “ace_tao-8.0.0-squished”.

Perry, I’ll admit to having an incomplete understanding of git, but I’m unlikely to pick up deeper understanding that I’ll rarely exercise (made it almost 20 years so far) so will try the cheating method suggested by Jonny.

Thanks all and will turn this around asap.

- Tom

> On Jun 2, 2024, at 8:03 AM, Johnny Willemsen ***@***.***> wrote:
> 
> 
> Or copy the changes files to a tmp dir, make a new bracht and make one commit with all changes files and open a new PR
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/macports/macports-ports/pull/24095#issuecomment-2143891145>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AANLDANXD32EHQGU2HECWVDZFMX4XAVCNFSM6AAAAABIGFXGT2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNBTHA4TCMJUGU>.
> You are receiving this because you were mentioned.
>
